### PR TITLE
Filter menu groups based on user access to child items

### LINKED ISSUE #3: Menu link groups show for users with no permissions to edit the sub menu items
When logged in as a user with only permission to edit a few content types, the links under add->content that they cannot access are hidden. 
When including in a menu group, the group links are shown even if there are no menu items they can access.

Screenshot 2020-09-18 at 9 23 57 am

### PR DESCRIPTION
**Pull request by andybroomfield**
Fix #3
Use hook_preprocess_menu to filter out menu link groups where the current
user does not have access to the child items.

### CODE REVIEW

**Comment by andybroomfield** [comment]
@Adnan-cds See issue #3 for the context. 
What I'm struggling with is creating a test to verify that a user cannot access any menu items in the group, they can't see the group. This works manually testing it though.

**Comment by Adnan-cds** [comment]
Hi Andy, I can confirm that the fix indeed works.  There are no cache related gaps in the fix.  I will try to add a Kernel test for this.  Failing that I will do the usual Functional test.  But that's for next week.  Hope that's okay.

**Comment by Adnan-cds** [comment]
I wonder why the Github actions test didn't run after my last commit :(

**Comment by andybroomfield** [comment]
This looks good, thanks @Adnan-cds.
As I did the original work, and Adnan wrote the test, someone else needs to review so we can merge.

File: localgov_menu_link_group.module [hunk 1]
```diff
@@ -9,6 +9,60 @@ declare(strict_types = 1);
 
 use Drupal\localgov_menu_link_group\MenuLinkGrouper;
 use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Menu\MenuTreeParameters;
+
+/**
+ * Implements hook_preprocess_menu().
+ */
+function localgov_menu_link_group_preprocess_menu(&$variables) {
+  $variables['menu_name'] = $variables['menu_name'] ?? 'admin';
```
Comment (andybroomfield) [inline localgov_menu_link_group.module:18]: I was going to query setting default menu name to admin, turns out it doesn't even matter what menu name gets passed, just needs a string and it still works :-).

**Comment by andybroomfield** [comment]
Be good to get this merged in time for the next release.

**Comment by stephen-cox** [comment]
Looks good. Just merged latest changes to run tests against D8 and D9. If they pass then I'm happy to approve.

**Review by stephen-cox — APPROVED** [review]
Tests passing - approved

**Comment by Adnan-cds** [comment]
Thanks Steve.  Merging...

**Pull request merged**